src/libstore/local-overlay-store.hh

```diff
@@ -124,6 +124,8 @@ private:
 
     void optimiseStore() override;
 
+    bool verifyAllValidPaths(RepairFlag repair, StorePathSet & validPaths) override;
```

Comment:
I've been thinking on all our overridden methods we should document what we did / how it differs from the vanilla `LocalStore` version.

```suggestion
    /**
     * Check all paths registered in the upper DB.
     *
     * Note that this includes store objects that reside in either overlayfs layer; just enumerating the contents of the upper layer would skip them.
     * 
     * We don't verify the contents of both layers on the assumption that the lower layer is far bigger, and also the observation that anything not in the upper db the overlayfs doesn't yet care about.  
     */
    bool verifyAllValidPaths(RepairFlag repair, StorePathSet & validPaths) override;
```

Here's a starter version, but feel free to reword and fix line length :)